src/algorithms/fardetectors/FarDetectorLinearTracking.cc

```diff
@@ -0,0 +1,177 @@
+                  ROOT::Math::XYZPoint((hit2-outPos)*layerWeights[2]).GetCoordinates(mc.GetMatrixArray());
+                  ROOT::Math::XYZPoint((hit3-outPos)*layerWeights[3]).GetCoordinates(md.GetMatrixArray());
+
+                  TMatrixD vecMatrix(3,4);
+                  vecMatrix.SetSub(0,0,ma);
+                  vecMatrix.SetSub(0,1,mb);
+                  vecMatrix.SetSub(0,2,mc);
+                  vecMatrix.SetSub(0,3,md);
+
+                  TDecompSVD decomp(vecMatrix.T());
+
+                  decomp.Decompose();
+
+                  auto decompVec = decomp.GetV().GetMatrixArray();
+
+                  auto varMatrix = vecMatrix*(decomp.GetV());
+                  auto subMat    = varMatrix.GetSub(0,3,1,2);
+
+                  auto subAsArray  = subMat.GetMatrixArray();
+                  ROOT::VecOps::RVec<double> subAsVector(subAsArray,subAsArray+8);
+                  double outChi2 = Sum(subAsVector*subAsVector)/8;
+
+                  // Chi2 cut on fit to cluster points
+                  if(outChi2>0.001) continue; // Optimise later or add as config
+
+                  ROOT::Math::XYZVector outVec(decompVec[0],decompVec[3],decompVec[6]);
+
+                  // Make sure fit was pointing in the right direction
+                  if(outVec.Z()>0) outVec*=-1;
+
+                  //position vector crosses x axis, swap to z exit later
+                  auto exitPos = outPos-(outPos.x()/outVec.x())*outVec;
+
+                  // Create track parameters edm4eic structure
+                  // TODO - populate more of the fields
+                  int type = 0;
+                  // Plane Point
+                  edm4hep::Vector2f loc(exitPos.y()*10,exitPos.z()*10); //Temp unit transform
+                  // Point Error
+                  edm4eic::Cov2f locError;
+                  float theta = outVec.Unit().Theta();
+                  float phi   = outVec.Unit().Phi();
+                  edm4eic::Cov2f dirError;
+                  float qOverP;
+                  edm4eic::Cov3f momentumError;
+                  float time      = 0;
+                  float timeError = 0;
+                  float charge    = 0;
+                  float path      = 0;
+                  float pathError = 0;
+
+                  edm4eic::TrackParameters outTrack(type,loc,locError,theta,phi,qOverP,momentumError,time,timeError,charge);
+                  outputTracks->push_back(outTrack);
+
+                  // Plane Point
+                  edm4hep::Vector3f pos(exitPos.x()*10,exitPos.y()*10,exitPos.z()*10); //Temp unit transform
+                  edm4eic::Cov3f    posError;
+
+
+                  edm4hep::Vector3f mom(outVec.x(),outVec.y(),outVec.z()); //Temp unit transform
+                  edm4eic::Cov3f    momError;
+
+
+//                edm4eic::TrackPoint outTrack(pos,posError,mom,momError,time,timeError,theta,phi,dirError,path,pathError);
+//                outputTracks->push_back(outTrack);
```

Comment:
See https://github.com/eic/EICrecon/pull/862/commits/f10c14fdab71cffefbfea5d28cfd98cf2dd79b7f for a robust solution to EDM4eic versioning.

Reply:
Or just remove the TrackPoint if you don't want to write it.